src/map/utils/battleutils.cpp

```diff
@@ -667,6 +659,14 @@ bool HandleSpikesDamage(CBattleEntity* PAttacker, CBattleEntity* PDefender, apAc
                 if (Action->reaction == REACTION_BLOCK)
                 {
                     PAttacker->addHP(-Action->spikesParam);
+
+                    // Subpower is the remaining damage that can be reflected. When it reaches 0 the effect ends
+                    int remainingReflect = PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_REPRISAL)->GetSubPower();

```

Comment:
What do you mean? like 
CStatusEffect reprisalEffect = PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_REPRISAL);
?